luxonis_train/attached_modules/losses/pml_loss.py

```diff
@@ -0,0 +1,122 @@
+import warnings
+
+from pytorch_metric_learning.losses import (
+    AngularLoss,
+    ArcFaceLoss,
+    CircleLoss,
+    ContrastiveLoss,
+    CosFaceLoss,
+    CrossBatchMemory,
+    DynamicSoftMarginLoss,
+    FastAPLoss,
+    GeneralizedLiftedStructureLoss,
+    HistogramLoss,
+    InstanceLoss,
+    IntraPairVarianceLoss,
+    LargeMarginSoftmaxLoss,
+    LiftedStructureLoss,
+    ManifoldLoss,
+    MarginLoss,
+    MultiSimilarityLoss,
+    NCALoss,
+    NormalizedSoftmaxLoss,
+    NPairsLoss,
+    NTXentLoss,
+    P2SGradLoss,
+    PNPLoss,
+    ProxyAnchorLoss,
+    ProxyNCALoss,
+    RankedListLoss,
+    SignalToNoiseRatioContrastiveLoss,
+    SoftTripleLoss,
+    SphereFaceLoss,
+    SubCenterArcFaceLoss,
+    SupConLoss,
+    TripletMarginLoss,
+    TupletMarginLoss,
+)
+from torch import Tensor
+
+from .base_loss import BaseLoss
+
+# Dictionary mapping string keys to loss classes
+loss_dict = {
+    "AngularLoss": AngularLoss,
+    "ArcFaceLoss": ArcFaceLoss,
+    "CircleLoss": CircleLoss,
+    "ContrastiveLoss": ContrastiveLoss,
+    "CosFaceLoss": CosFaceLoss,
+    "DynamicSoftMarginLoss": DynamicSoftMarginLoss,
+    "FastAPLoss": FastAPLoss,
+    "GeneralizedLiftedStructureLoss": GeneralizedLiftedStructureLoss,
+    "InstanceLoss": InstanceLoss,
+    "HistogramLoss": HistogramLoss,
+    "IntraPairVarianceLoss": IntraPairVarianceLoss,
+    "LargeMarginSoftmaxLoss": LargeMarginSoftmaxLoss,
+    "LiftedStructureLoss": LiftedStructureLoss,
+    "ManifoldLoss": ManifoldLoss,
+    "MarginLoss": MarginLoss,
+    "MultiSimilarityLoss": MultiSimilarityLoss,
+    "NCALoss": NCALoss,
+    "NormalizedSoftmaxLoss": NormalizedSoftmaxLoss,
+    "NPairsLoss": NPairsLoss,
+    "NTXentLoss": NTXentLoss,
+    "P2SGradLoss": P2SGradLoss,
+    "PNPLoss": PNPLoss,
+    "ProxyAnchorLoss": ProxyAnchorLoss,
+    "ProxyNCALoss": ProxyNCALoss,
+    "RankedListLoss": RankedListLoss,
+    "SignalToNoiseRatioContrastiveLoss": SignalToNoiseRatioContrastiveLoss,
+    "SoftTripleLoss": SoftTripleLoss,
+    "SphereFaceLoss": SphereFaceLoss,
+    "SubCenterArcFaceLoss": SubCenterArcFaceLoss,
+    "SupConLoss": SupConLoss,
+    "TripletMarginLoss": TripletMarginLoss,
+    "TupletMarginLoss": TupletMarginLoss,
+}
+
+
+class MetricLearningLoss(BaseLoss):
```

Comment:
All the losses here are for embedding task right? And this class is a wrapper for all of them? Maybe we change the name so it reflects this more e.g. `EmbeddingLossWrapper` (probably there is a better name)